### PR TITLE
Add upload of installation script logs

### PR DESCRIPTION
**Pull request by ccressent**
Relocating the work from https://github.com/SumoLogic/sumologic-otel-collector/pull/1312 into this repository now that the installation scripts have moved.

### CODE REVIEW

File: install-script/install.sh [hunk 1]
```diff
@@ -167,6 +178,17 @@ Supported env variables:
 EOF
 }
 
+# shellcheck disable=SC2317  # Don't warn about unreachable commands in this function
+# ShellCheck may incorrectly believe that code is unreachable if it's invoked in
+# a trap, like the reporter function.
+function reporter {
+    if ! $DISABLE_INSTALLATION_TELEMETRY; then
+        echo "SUMOLOGIC_INSTALLATION_TOKEN=${SUMOLOGIC_INSTALLATION_TOKEN}" >> "$INSTALLATION_LOGFILE"
+        curl --silent --location -X POST --data-binary @"${INSTALLATION_LOGFILE}" "${INSTALLATION_LOGFILE_ENDPOINT}"
```
Comment (swiatekm-sumo) [inline install-script/install.sh:187]: We should really delete the logfile here, if we put the installation token in it.

File: install-script/install.ps1 [hunk 2]
```diff
@@ -396,19 +404,48 @@ function Get-BinaryFromUri {
...
 # Main code
 ##
 
 try {
-    if ($InstallationToken -eq $null -or $InstallationToken -eq "") {
-        Write-Error "Installation token has not been provided. Please set the SUMOLOGIC_INSTALLATION_TOKEN environment variable." -ErrorAction Stop
+    $InstallationLogFile = New-TemporaryFile
+
+    if ($InstallationLogFileEndpoint -eq "") {
+        $InstallationLogFileEndpoint = "https://stag-open-events.sumologic.net/api/v1/collector/installation/logs"
```
Comment (swiatekm-sumo) [inline install-script/install.ps1:445]: This endpoint is not publicly accessible, as far as I know. Is it really what we should be using here?

**Review by swiatekm-sumo — COMMENTED** [review]
If I understand correctly, these routines are supposed to fail quietly if sending fails, which is why tests pass. Can we add a test which starts a small webserver, sets the endpoint to that webserver, and then asserts that it got some data?

File: install-script/install.ps1 (hunk 2 above)
Comment (ccressent) [reply install-script/install.ps1:445]: The endpoint is actually publicly accessible (be it for staging or production), else we wouldn't be able to send installation logs to it from wherever the collector gets installed. The URL is currently set to the staging endpoint, but I'll change it to the production one.

The backend team went through a separate security review and subsequent changes to make sure that this endpoint was OK to have as publicly accessible.

**Comment by ccressent** [comment]
@swiatekm-sumo 
Are we good to merge this?

File: install-script/install.sh [hunk 3]
```diff
@@ -131,6 +137,10 @@ CURL_MAX_TIME=1800
 # set by check_dependencies therefore cannot be set by set_defaults
 SYSTEMD_DISABLED=false
 
+DISABLE_INSTALLATION_TELEMETRY=false
+INSTALLATION_LOGFILE="/tmp/sumologic-otel-collector_installation.log"
```
Comment (swiatekm-sumo) [inline install-script/install.sh:141]: You should use `${TMPDIR}` as the base instead. That has a temp directory with the right permissions on all supported systems.

**Review by swiatekm-sumo — APPROVED** [review]
One comment about temp file placement on Linux and Darwin, LGTM otherwise.

File: install-script/install.sh (hunk 3 above)
Comment (ccressent) [reply install-script/install.sh:141]: 100% agree ; good catch

**Pull request merged**